dependencies.go

```diff
@@ -116,6 +125,20 @@ func (d faultyDiskDependency) create(path string) (file, error) {
 	}
 	return d.newFaultyFile(f), nil
 }
+func (d faultyDiskDependency) remove(path string) error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	if !*d.disabled {
```

Comment:
What is the difference between disabled and failed? In this function, you sometimes set 'failed' equal to true, but that doesn't impact the operation of this function.

Reply:
if `failed` is `true` every sync will fail. It is set to true once the first write failed.
`disabled` can be used to "fix" the disk temporarily to avoid corruptions.